packages/chrome/src/elements/sheet/Sheet.spec.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useState, forwardRef } from 'react';
+import userEvent from '@testing-library/user-event';
+import { render } from 'garden-test-utils';
+
+import { Sheet, ISheetProps } from './Sheet';
+
+describe('Sheet', () => {
+  const Example = forwardRef<HTMLElement, ISheetProps>((props: ISheetProps, ref) => {
+    const [isOpen, setIsOpen] = useState(props.isOpen);
+    const buttonRef = useRef<HTMLButtonElement>(null);
+
+    return (
+      <>
+        <button ref={buttonRef} onClick={() => setIsOpen(!isOpen)}>
+          Toggle Sheet
+        </button>
+        <Sheet ref={ref} isOpen={isOpen} {...props}>
+          <Sheet.Header>
+            <Sheet.Title>title</Sheet.Title>
+            <Sheet.Description>description</Sheet.Description>
+          </Sheet.Header>
+          <Sheet.Body>body</Sheet.Body>
+          <Sheet.Footer>
+            <Sheet.FooterItem>
+              <button>button1</button>
+            </Sheet.FooterItem>
+            <Sheet.FooterItem>
+              <button>button2</button>
+            </Sheet.FooterItem>
+          </Sheet.Footer>
+          <Sheet.Close />
+        </Sheet>
+      </>
+    );
+  });
+
+  Example.displayName = 'Example';
+
+  it('passes ref to underlying DOM element', () => {
+    const ref = React.createRef<HTMLDivElement>();
+    const { getByRole, getByText } = render(<Example ref={ref} />);
+
+    userEvent.click(getByText('Toggle Sheet'));
+
+    expect(getByRole('complementary')).toBe(ref.current);
+  });
+
+  it('is only mounted when isOpen is true', () => {
```

Comment:
Not a blocker, but most tests (testing from user perspective) should have declarative descriptions. Maybe some variant of `opens and closes sheet`.

Note: there are times for imperative descriptions — when Garden tests implementation details. Library authors occasionally have to test implementation details. For those tests, imperative descriptions make more sense.

Reply:
Noted! I think I follow, could you give an example of what you mean by imperative tests?

Reply:
Yeah for sure! I meant "imperative description" and not "imperative tests". Here's an [example](https://github.com/zendeskgarden/react-containers/blob/main/packages/grid/src/GridContainer.spec.tsx#L411) of testing implementation details. Imperative descriptions are written from the perspective of the developer (not end user) and typically contain programming phrases such as `isOpen`, `true`, `onChange`, "invokes", "calls", etc.